src/zcash/Note.cpp

```diff
@@ -38,6 +40,55 @@ uint256 SproutNote::nullifier(const SproutSpendingKey& a_sk) const {
     return PRF_nf(a_sk, rho);
 }
 
+// Construct and populate Sapling note for a given payment address and value.
+SaplingNote::SaplingNote(const SaplingPaymentAddress& address, const uint64_t value) : BaseNote(value) {
+    d = address.d;
+    pk_d = address.pk_d;
+    librustzcash_sapling_generate_r(r.begin());
+}
+
+// Call librustzcash to compute the commitment
+boost::optional<uint256> SaplingNote::cm() const {
+    uint256 result;
+    if (!librustzcash_sapling_compute_cm(
+            d.data(),
+            pk_d.begin(),
+            value(),
+            r.begin(),
+            result.begin()
+        ))
+    {
+        return boost::optional<uint256>();
+    }
+
+    return result;
+}
+
+// Call librustzcash to compute the nullifier
+boost::optional<uint256> SaplingNote::nullifier(const SaplingSpendingKey& sk, const uint64_t position) const
+{
+    auto vk = sk.full_viewing_key();
+    auto ak = vk.ak;
+    auto nk = vk.nk;
+
+    uint256 result;
+    if (!librustzcash_sapling_compute_nf(
+            d.data(),
+            pk_d.begin(),
+            value(),
+            r.begin(),
+            ak.begin(),
+            nk.begin(),
+            position,
+            result.begin()
+    ))
+    {
+        return boost::optional<uint256>();
```

Comment:
I thought there was a `boost::none` or something you can use instead. Not blocking.

Reply:
Yes, `return boost::none;` works.

Reply:
Equivalent.  Updated to use boost::none as its easier on the eyes.